src/extractor/guidance/sliproad_handler.cpp

```diff
@@ -29,273 +32,720 @@ SliproadHandler::SliproadHandler(const IntersectionGenerator &intersection_gener
+
+            const auto length = haversineDistance(node_info_list[intersection_node_id], //
+                                                  node_info_list[next->node]);
+
+            const double perpendicular_angle = 90 + FUZZY_ANGLE_DIFFERENCE;
+
+            if (length >= MIN_LENGTH)
+            {
+                if (!snuggles)
+                    continue;
+
+                if (deviation_from_straight > perpendicular_angle)
+                    continue;
+            }
+        }
+
+        // Check for area under triangle `bdc`.
+        //
+        // a ... b .... c .
+        //       `      .
+        //         `    .
+        //           `  .
+        //              d
+        //
+        const auto area_threshold = std::pow(
+            scaledThresholdByRoadClass(MAX_SLIPROAD_THRESHOLD, sliproad_data.road_classification),
+            2.);
+
+        if (!isValidSliproadArea(
+                area_threshold, intersection_node_id, next->node, sliproad_edge_target))
+        {
+            continue;
+        }
+
+        // Check all roads at `d` if one is connected to `c`, is so `bd` is Sliproad.
+        for (const auto &candidate_road : target_intersection)
+        {
+            const auto &candidate_data = node_based_graph.GetEdgeData(candidate_road.eid);
+
+            // Name mismatch: check roads at `c` and `d` for same name
+            const auto name_mismatch = [&](const NameID road_name_id) {
+                return util::guidance::requiresNameAnnounced(road_name_id,              //
+                                                             candidate_data.name_id,    //
+                                                             name_table,                //
+                                                             street_name_suffix_table); //
+            };
+
+            const auto candidate_road_name_mismatch = std::all_of(begin(target_road_name_ids), //
+                                                                  end(target_road_name_ids),   //
+                                                                  name_mismatch);              //
+
+            if (candidate_road_name_mismatch)
                 continue;
 
-            for (const auto &candidate_road : target_intersection)
+            // If the Sliproad `bd` is a link, `bc` and `cd` must not be links.
+            if (!isValidSliproadLink(sliproad, next_road, candidate_road))
+                continue;
+
+            if (node_based_graph.GetTarget(candidate_road.eid) == next->node)
+            {
+                sliproad.instruction.type = TurnType::Sliproad;
+                sliproad_found = true;
+                break;
+            }
+            else
             {
-                const auto &candidate_data = node_based_graph.GetEdgeData(candidate_road.eid);
-                if (target_road_names.count(candidate_data.name_id) > 0)
+                const auto skip_traffic_light_intersection = intersection_generator(
```

Comment:
should probably also use `GetActualNextIntersection` from the intersection Generator